src/Hazelcast.Net/Serialization/Compact/SchemaBuilderWriter.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Hazelcast.Models;
+
+namespace Hazelcast.Serialization.Compact
+{
+    internal class SchemaBuilderWriter : ICompactWriter
+    {
+        private readonly string _typeName;
+        private readonly List<SchemaField> _fields = new();
+
+        public SchemaBuilderWriter(string typeName)
+        {
+            _typeName = typeName;
+        }
+
+        public FieldKind GetFieldKind(string name)
```

Comment:
What's the goal here?

Reply:
`SchemaBuilderWriter` is a `ICompactWriter` but a special one: we pretend to serialize an object through it and it collects the fields and produces the corresponding schema. But it does not serialize anything really. So although it's a `ICompactWriter` you cannot use it as a true writer. Some methods are not supported. That one (`GetFieldKind`) cannot be supported since we don't know the schema beforehand. But... since it's defined on the interface, we have to implement it - and throw.